app/models/event_schedule.rb

```diff
@@ -12,6 +12,7 @@ class EventSchedule < ActiveRecord::Base
   scope :confirmed, -> { joins(:event).where('state = ?', 'confirmed') }
   scope :canceled, -> { joins(:event).where('state = ?', 'canceled') }
   scope :withdrawn, -> { joins(:event).where('state = ?', 'withdrawn') }
+  scope :current, -> { where('start_time <= ?', Time.current).map { |es| es if es.end_time >= Time.current }.compact }
```

Comment:
why not using `select` instead of `map`? 

`scope :current, -> { where('start_time <= ?', Time.current).select { |es| es.end_time >= Time.current } }`